src/mscorlib/src/System/String.cs

```diff
@@ -552,10 +549,13 @@ private unsafe static int CompareOrdinalHelper(String strA, String strB)
                 DiffOffset8: a += 4; b += 4;
                 DiffOffset4: a += 4; b += 4;
 #else
-                DiffOffset8: a += 2; b += 2;
-                DiffOffset6: a += 2; b += 2;
-                DiffOffset4: a += 2; b += 2;
-                DiffOffset2: a += 2; b += 2;
+                // Use jumps instead of falling through, since
+                // otherwise going to DiffOffset8 will involve
+                // 8 add instructions before getting to DiffNextInt
+                DiffOffset8: a += 8; b += 8; goto DiffOffset0;
+                DiffOffset6: a += 6; b += 6; goto DiffOffset0;
+                DiffOffset4: a += 4; b += 4; goto DiffOffset0;
```

Comment:
@jkotas OK, so I was finally able to procure measurements for this. Falling through is definitely faster than jumping for both `DiffOffset4` on x86 and `DiffOffset8` on x64, however interestingly jumping seems to be faster for `DiffOffset6` and for `DiffOffset8` they seem to be about the same.

- [jumps](https://gist.github.com/jamesqo/75d45956d7430b5bdf9ebd2b1c978560)
- [fallthrough](https://gist.github.com/jamesqo/5032ae49dee49b56856e56fab9d040fb)
- [test source](https://gist.github.com/jamesqo/17ef7ef14136a624378d77f070bed08b)